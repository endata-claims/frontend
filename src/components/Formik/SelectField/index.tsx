import React from 'react'

import TextField, { FormikTextFieldProps } from '../TextField'
import { MenuItem } from '@material-ui/core'

export type FormikSelectFieldProps = FormikTextFieldProps & {
  options?: SelectOption[]
}
export type SelectOption = {
  label?: string
  value: string
}

const FormikSelectField: React.FC<FormikSelectFieldProps> = ({ options, ...props }) => {
  return (
    <TextField {...props} select>
      {options
        ? options.map(({ label, value }) => (
          <MenuItem key={value} value={value}>
            {label}
          </MenuItem>
        ))
        : (
          <MenuItem disabled>
            No option
          </MenuItem>
        )
      }
    </TextField>
  )
}
export default FormikSelectField