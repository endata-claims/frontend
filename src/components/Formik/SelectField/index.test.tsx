import React from 'react'
import { shallow } from 'enzyme'
import { Formik, Form } from 'formik'
import YouTubeIcon from '@material-ui/icons/YouTube'

import SelectField from '.'

describe('SelectField tests', () => {
  const renderWrapper = (props = {}) => shallow(
    <Formik initialValues={{ select: '' }}
      onSubmit={values => console.log(values)}
    >
      <Form>
        <SelectField label='Text field' name='text' {...props} />
      </Form>
    </Formik>
  )

  describe('snapshots', () => {
    it('should render without options', () => {
      expect(renderWrapper()).toMatchSnapshot()
    })
    it('should render with options', () => {
      const options = [
        { label: 'Red', value: 'red' },
        { label: 'Blue', value: 'blue' },
      ]
      expect(renderWrapper({ options })).toMatchSnapshot()
    })
    it('should render with icon', () => {
      expect(renderWrapper({
        startAdornment: <YouTubeIcon />
      })).toMatchSnapshot()
    })
    it('should render with right icon', () => {
      expect(renderWrapper({
        endAdornment: <YouTubeIcon />
      })).toMatchSnapshot()
    })
  })
})