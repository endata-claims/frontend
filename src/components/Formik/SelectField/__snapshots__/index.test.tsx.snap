// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField tests snapshots should render with icon 1`] = `ShallowWrapper {}`;

exports[`SelectField tests snapshots should render with options 1`] = `ShallowWrapper {}`;

exports[`SelectField tests snapshots should render with right icon 1`] = `ShallowWrapper {}`;

exports[`SelectField tests snapshots should render without options 1`] = `ShallowWrapper {}`;
